website/docs/providers/notifier_provider/todos/raw.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+/* SNIPPET START */
+
+// The state of our Notifier should be immutable.
+// We could also use packages like Freezed to help with the implementation.
+@immutable
+class Todo {
+  const Todo({
+    required this.id,
+    required this.description,
+    required this.completed,
+  });
+
+  // All properties should be `final` on our class.
+  final String id;
+  final String description;
+  final bool completed;
+
+  // Since Todo is immutable, we implement a method that allows cloning the
+  // Todo with slightly different content.
+  Todo copyWith({String? id, String? description, bool? completed}) {
+    return Todo(
+      id: id ?? this.id,
+      description: description ?? this.description,
+      completed: completed ?? this.completed,
+    );
+  }
+}
+
+// The Notifier class that will be passed to our NotifierProvider.
+// This class should not expose state outside of its "state" property, which means
+// no public getters/properties!
+// The public methods on this class will be what allow the UI to modify the state.
+class TodosNotifier extends Notifier<List<Todo>> {
+  // We initialize the list of todos to an empty list
+  @override
+  List<Todo> build() {
```

Comment:
I think it's a good opportunity to use AsyncNotifier, and have `build` do a network request

Reply:
Haven't addressed it yet.
Do you think we should add a new example for AsyncNotifier on top of the Notifier example or just replace it?

Reply:
Added an example with the AsyncNotifier.